importer/Bosnia.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Self-contained-ish module for Bosnia/Herzegovina.
+
+Since this data does not come from WLM,
+this module contains its own little
+importer that parses the csv data
+(in several files in a subdirectory called
+upon with --datadir).
+"""
+import argparse
+import os
+import dateparser
+
+import wikidataStuff.wdqsLookup as lookup
+
+from Monument import Monument
+from Uploader import Uploader
+import importer_utils as utils
+
+MAPPINGS_DIR = "mappings"
+REPORTING_DIR = "reports"
+PREVIEW_DIR = "previews"
+
+
+class BosniaMonument(Monument):
+
+    def __init__(self, db_row_dict, mapping, data_files, existing, repository):
+        """
+        Initialize the data object.
+
+        Like Monument.py but with some simplifications
+        re: attribute setting
+        """
+        self.raw_data = db_row_dict
+        self.props = data_files["_static"]["props"]
+        self.adm0 = data_files["_static"]["adm0"]
+        self.sources = data_files["_static"]["sources"]
+        self.common_items = data_files["_static"]["common_items"]
+        for k, v in db_row_dict.items():
+            setattr(self, k.lower().strip().replace(" ", "_"), v)
+        self.data_files = data_files
+        self.construct_wd_item(mapping)
+        self.existing = existing
+        self.repo = repository
+        self.problem_report = {}
+
+    def create_source(self):
+        source_item = "Q42750314"
+        last_edit = "2017-09-08"
+        timestamp = utils.date_to_dict(last_edit, "%Y-%m-%d")
+        prop_stated = self.props["stated_in"]
+        prop_date = self.props["publication_date"]
```

Comment:
Should this actually be "retrieval date" since the google doc isn't the actual date of publication?

Reply:
:white_check_mark: this actually required hacking the Uploader.... :open_mouth:

Reply:
...though of course we don't _really_ know how accurate this date is, regardless of what prop we stick on it....